toolkit/offchain/tests/integration_tests.rs

```diff
@@ -407,3 +439,64 @@ async fn assert_reserve_deposited<T: QueryLedgerState>(
 	)
 	.await;
 }
+
+async fn assert_mutable_settings_eq<T: QueryLedgerState + ogmios_client::OgmiosClient>(
+	genesis_utxo: UtxoId,
+	updated_mutable_settings: ReserveMutableSettings,
+	client: &T,
+) {
+	let (_reserve_utxo, reserve_settings) = get_reserve_utxo(genesis_utxo, client).await;
+
+	let mutable_settings = reserve_settings.mutable_settings;
+	assert_eq!(
+		mutable_settings.total_accrued_function_script_hash,
+		updated_mutable_settings.total_accrued_function_script_hash
+	);
+	assert_eq!(mutable_settings.initial_incentive, updated_mutable_settings.initial_incentive);
+}
+
+async fn get_reserve_utxo<T: QueryLedgerState + ogmios_client::OgmiosClient>(
+	genesis_utxo: UtxoId,
+	client: &T,
+) -> (OgmiosUtxo, ReserveDatum) {
+	let network = client.shelley_genesis_configuration().await.unwrap().network.to_csl();
+
+	let validator_script = PlutusScript::from_wrapped_cbor(
+		raw_scripts::VERSION_ORACLE_VALIDATOR,
+		Language::new_plutus_v2(),
+	)
+	.unwrap()
+	.apply_data(genesis_utxo)
+	.unwrap();
+	let policy_script = PlutusScript::from_wrapped_cbor(
+		raw_scripts::VERSION_ORACLE_POLICY,
+		Language::new_plutus_v2(),
+	)
+	.unwrap()
+	.apply_data(genesis_utxo)
+	.unwrap()
+	.apply_uplc_data(validator_script.address_data(network).unwrap())
+	.unwrap();
+
+	let validator =
+		PlutusScript::from_wrapped_cbor(raw_scripts::RESERVE_VALIDATOR, Language::new_plutus_v2())
+			.unwrap()
+			.apply_uplc_data(uplc::PlutusData::BoundedBytes(
+				policy_script.script_hash().to_vec().into(),
+			))
+			.unwrap();
+
+	let validator_address = validator.address(network).to_bech32(None).unwrap();
```

Comment:
```suggestion
        let data = scripts_data::get_scripts_data_with_ogmios(genesis_utxo, client).await.unwrap();
	let validator_address = data.addresses.reserve_validator;
```
Requires tiny refactor in get_scripts_data, so it accepts `&impl QueryNetwork` and added bound to this `get_reserve_utxo`.

Reply:
Thank you for this!